pkgs/tools/misc/heimdall/default.nix

```diff
@@ -1,46 +1,44 @@
 {
   lib,
   stdenv,
-  mkDerivation,
-  fetchFromGitHub,
+  fetchFromSourcehut,
   cmake,
-  zlib,
   libusb1,
+  pkg-config,
+  wrapQtAppsHook,
+  zlib,
   enableGUI ? false,
   qtbase ? null,
 }:
 
-mkDerivation rec {
+stdenv.mkDerivation rec {
   pname = "heimdall${lib.optionalString enableGUI "-gui"}";
-  version = "1.4.2";
+  version = "2.2.1";
 
-  src = fetchFromGitHub {
-    owner = "Benjamin-Dobell";
+  src = fetchFromSourcehut {
+    owner = "~grimler";
     repo = "Heimdall";
     rev = "v${version}";
-    sha256 = "1ygn4snvcmi98rgldgxf5hwm7zzi1zcsihfvm6awf9s6mpcjzbqz";
+    sha256 = "sha256-x+mDTT+oUJ4ffZOmn+UDk3+YE5IevXM8jSxLKhGxXSM=";
   };
 
   buildInputs = [
-    zlib
     libusb1
-  ] ++ lib.optional enableGUI qtbase;
-  nativeBuildInputs = [ cmake ];
+    zlib
+  ] ++ lib.lists.optionals enableGUI [
+    qtbase
+    wrapQtAppsHook
+  ];
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
 
   cmakeFlags = [
     "-DDISABLE_FRONTEND=${if enableGUI then "OFF" else "ON"}"
     "-DLIBUSB_LIBRARY=${libusb1}"
   ];
 
-  preConfigure =
-    ''
-      # Give ownership of the Galaxy S USB device to the logged in user.
-      substituteInPlace heimdall/60-heimdall.rules --replace 'MODE="0666"' 'TAG+="uaccess"'
-    ''
-    + lib.optionalString stdenv.hostPlatform.isDarwin ''
-      substituteInPlace libpit/CMakeLists.txt --replace "-std=gnu++11" ""
```

Comment:
Removal of this is currently untested, but the old replace pattern does not apply anymore anyway.